### PR TITLE
Hide future posts/links and auto publish them on speficied date

### LINKED ISSUE #147: Add automatic publish-on-date functionality
Mostly for events, but auto publish is something that would be a nice to have for any kind of content in the blog.

#### Editor flow

The main flow from an editor perspective would be the same as creating a normal post to publish with the small difference of having an extra field to fill "Publishing Date" for example that would indicate _when_ the post would be published (timezones can be configured to be either the content editor's or always UTC, see docs [here](https://forestry.io/docs/settings/fields/datetime/)).

We already have a "Date" field, but having a separate one for publishing would allow for us to support use cases like: I add an event that happens in 3 weeks but I want to publish its link 2 weeks before the event.

#### Implementation

- On forestry we need to set the new "Publishing Date" field with the timezone configuration (either the content editor's timezone or always UTC) and then add a description to the field input referencing the config so that content editors know which timezone is being considered when they are inputting the date and time.
- The "Publishing Date" would be used to either include a post in the build or not. If it's not the time yet to publish, it wouldn't be included and the link would result in a 404. Careful take into account all of the use cases (blog post listing, sitemap, and rss feed – index.json for ipfs.io is derived from rss feed).
- The bulk of the work would be to auto publish the website with the new entry, meaning simply trigger a deployment of the production fleek app.
Fleek provides a [CLI](https://docs.fleek.co/fleek-cli/overview/) which we can use to trigger a deployment programmatically.
- To correctly trigger the publish we can run an action daily to check for posts to publish on the current day (it will run unnecessarily most of the days, but since this is public repo, we don't pay for it – the limit until we pay is [72h](https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#usage-limits) which will be very hard to reach).
   - we could increase the frequency in order to have more accurate publishing considering that running only once a day would mean we could publish an event 23 hours and 59mins too late? Maybe twice or thrice a day?

### PR DESCRIPTION
**Pull request by zebateira**
This PR closes #147 

The schedule for the checks is 9am and 9pm UTC everyday. @jessicaschilling does this make sense? We can change it to what makes more sense for posts/links to be published.

#### Notes

- I couldn't remove the routes for the hidden posts/links in the build process, so I just added a `hidden` field to each page. Then I use that field to hide items from the homepage and to redirect post routes to the 404 page (does not provide an actual 404 http code to the client). If you have another idea @cwaring @jdiogopeixoto feel free to pitch in.

#### Other changes that I added in this PR:

- `hidden` field can now be added to any page metadata (frontmatter) to be hidden
- Routes for link categories (events, videos, tutorials, etc) like `/events` will now redirect to the `404` page.

#### TODO

- [x] Remove WIP notice on forestry for publishing dates

### CODE REVIEW

File: .forestry/front_matter/templates/newslinks.yml [hunk 1]
```diff
@@ -35,17 +35,23 @@ fields:
...
-  - name: publishing_date
+  - name: publish_date
     type: datetime
-    label: Publishing Date
-    description: When the link should be published.<br>Use this to schedule publishing
-      the link in the future.<br>If you don't want to schedule, just ignore this.
+    label: Publish Date
+    description: '<strong>(WIP: Auto publishing feature is in the works)<br></strong><br>When
+      the link should be published.<br>Use this to schedule publishing the link in
+      the future.<br>If you don''t want to schedule, just ignore this.<br><br><strong><em>WARNING</em></strong><em>:
```
Comment (jessicaschilling) [inline .forestry/front_matter/templates/newslinks.yml:43]:
```suggestion
      the future.<br>If you don't want to schedule, leave this blank.<br><br><strong><em>WARNING</em></strong><em>:
```

File: .forestry/front_matter/templates/newslinks.yml [hunk 2]
```diff
@@ -35,17 +35,23 @@ fields:
...
     description: For <strong>events</strong>, use the date of the event itself.<br>For
       all other link types, use the date the original item was published.
-  - name: publishing_date
+  - name: publish_date
     type: datetime
-    label: Publishing Date
-    description: When the link should be published.<br>Use this to schedule publishing
-      the link in the future.<br>If you don't want to schedule, just ignore this.
+    label: Publish Date
+    description: '<strong>(WIP: Auto publishing feature is in the works)<br></strong><br>When
```
Comment (jessicaschilling) [inline .forestry/front_matter/templates/newslinks.yml:41]:
```suggestion
    description: 'When
```

File: .forestry/front_matter/templates/newslinks.yml [hunk 3]
```diff
@@ -35,17 +35,23 @@ fields:
...
+      the link should be published.<br>Use this to schedule publishing the link in
+      the future.<br>If you don''t want to schedule, just ignore this.<br><br><strong><em>WARNING</em></strong><em>:
+      Data is saved in the </em><a href="https://github.com/ipfs/ipfs-blog" title=""
+      target="_blank"><em>ipfs/ipfs-blog</em></a><em> public github repo as soon as
```
Comment (jessicaschilling) [inline .forestry/front_matter/templates/newslinks.yml:45]:
```suggestion
      target="_blank"><em>ipfs/ipfs-blog</em></a><em> public GitHub repo as soon as
```

File: .forestry/front_matter/templates/newslinks.yml [hunk 4]
```diff
@@ -35,17 +35,23 @@ fields:
...
+      you hit the save button. So any sensitive information will also be publicly
```
Comment (jessicaschilling) [inline .forestry/front_matter/templates/newslinks.yml:46]:
```suggestion
      you hit the Save button. This means anything you save will also be publicly
```

File: .forestry/front_matter/templates/newslinks.yml [hunk 5]
```diff
@@ -35,17 +35,23 @@ fields:
...
+      available on github, but hidden from the website blog.ipfs.io.</em>'
```
Comment (jessicaschilling) [inline .forestry/front_matter/templates/newslinks.yml:47]:
```suggestion
      visible on GitHub, but not on the blog.ipfs.io site. Be careful if you are drafting something sensitive!</em>'
```

**Review by jessicaschilling — CHANGES_REQUESTED** [review]
Nice! Added a few suggested text edits for the Forestry side of things. One other note and one question:

- Most posts are going to be drafted by folks in U.S. Pacific, so it probably makes sense to run the checks at times that work closer to their hours. 4:00 UTC is 9:00 Pacific - sound OK to you?

- > Routes for link categories (events, videos, tutorials, etc) like /events will now redirect to the 404 page.

Why this?

**Comment by cwaring** [comment]
Could I suggest that we add a date field to the links too and auto populate it with the current date? It might be useful to know when they are added plus it make the interface for editors the same across all content types

**Comment by zebateira** [comment]
> Most posts are going to be drafted by folks in U.S. Pacific, so it probably makes sense to run the checks at times that work closer to their hours. 4:00 UTC is 9:00 Pacific - sound OK to you?

Yap 👍  So 4am and 4pm UTC then?


> Routes for link categories (events, videos, tutorials, etc) like /events will now redirect to the 404 page.
> Why this?

It was a fix at an arm's reach so I just included it in this.
We wanted to address this issue before. ATM they are not rendering anything: https://blog.ipfs.io/events/
We might implement another fix in the future, but I think at least this small fix is welcomed.

**Comment by jessicaschilling** [comment]
@cwaring The two dates are to enable us to post links to events that will happen in the near future, as a means of promoting them. Primary example of this is the monthly meetup.

@zebateira Yes please on 4am/pm UTC 😊  ... regarding link category URLs, could we not route them to their respective filter URLs, e.g. `https://blog.ipfs.io/?category=Blog%20post`?

**Comment by zebateira** [comment]
@cwaring 

> Could I suggest that we add a date field to the links too and auto populate it with the current date? It might be useful to know when they are added plus it make the interface for editors the same across all content types.
>
> Having date and publish_date feels a bit odd - unless there is another reason for this that I'm missing?

Links already have a date field that is used for the date to display in the UI.
Only for events is the publish_date relevant: the "date" field for the events is the actual date when the event will happen, so we need to make a distinction there. As a bonus, this enables the other types to also have scheduling capabilities.

**Comment by zebateira** [comment]
> regarding link category URLs, could we not route them to their respective filter URLs, e.g. https://blog.ipfs.io/?category=Blog%20post?

Oh ok, will try this yeah 👍

**Comment by zebateira** [comment]
will defer the category pages redirect to #154 since the values will change soon.

File: .forestry/front_matter/templates/blog-post.yml [hunk 6]
```diff
@@ -27,23 +27,17 @@ fields:
 - type: datetime
   name: date
   label: Date
-  description: '"Published on" date. Remember this is <em>only</em> a list date for
-    display; posts are not automatically published!'
+  description: '"Published on" date.<br><br><strong>(WIP: Auto publishing feature
+    is in the works)</strong><br>To schedule this post for future publishing, simply
```
Comment (jdiogopeixoto) [inline .forestry/front_matter/templates/blog-post.yml:31]: Shouldn't we remove this WIP notice?

File: .forestry/front_matter/templates/blog-post.yml [hunk 7]
```diff
@@ -27,23 +27,17 @@ fields:
...
+    set the date to the future.<br><br><strong><em>WARNING</em></strong><em>: Data
+    is saved in the </em><a href="https://github.com/ipfs/ipfs-blog" title="" target="_blank"><em>ipfs/ipfs-blog</em></a><em>
+    public github repo as soon as you hit the save button. So any sensitive information
```
Comment (jdiogopeixoto) [inline .forestry/front_matter/templates/blog-post.yml:34]:
```suggestion
    public GitHub repo as soon as you hit the save button. So any sensitive information
```

File: .forestry/front_matter/templates/blog-post.yml [hunk 8]
```diff
@@ -27,23 +27,17 @@ fields:
...
+    will also be publicly available on github, but hidden from the website blog.ipfs.io.</em>'
```
Comment (jdiogopeixoto) [inline .forestry/front_matter/templates/blog-post.yml:35]:
```suggestion
    will also be publicly available on GitHub, but hidden from the website blog.ipfs.io.</em>'
```

**Review by jdiogopeixoto — CHANGES_REQUESTED** [review]
This is amazing, good job! 👏 

Just left some small comments.

File: .forestry/front_matter/templates/blog-post.yml (hunk 6 above)
Comment (zebateira) [inline .forestry/front_matter/templates/blog-post.yml:31]:
```suggestion
  description: '"Published on" date.<br><br><strong><br>To schedule this post for future publishing, simply
```

**Review by jdiogopeixoto — APPROVED** [review]

**Pull request merged**